capstone/src/test/java/com/google/sps/servlets/CreateGroupServletTest.java

```diff
@@ -0,0 +1,175 @@
+                          /* interests= */ new ArrayList<String>());
+
+  private static final Group GROUP_1 = new Group(
+                          /* memberIds= */ new ArrayList<String>(Arrays.asList(USER_ID)),
+                          /* challenges= */ new ArrayList<Long>(),
+                          /* posts= */ new ArrayList<Long>(),
+                          /* options= */ new ArrayList<Long>(),
+                          /* groupName= */ GROUP_NAME,
+                          /* headerImg= */ "",
+                          /* groupId= */ GROUP_ID);
+  
+  private static final Group GROUP_2 = new Group(
+                          /* memberIds= */ new ArrayList<String>(Arrays.asList(USER_ID)),
+                          /* challenges= */ new ArrayList<Long>(),
+                          /* posts= */ new ArrayList<Long>(),
+                          /* options= */ new ArrayList<Long>(),
+                          /* groupName= */ GROUP_NAME,
+                          /* headerImg= */ "",
+                          /* groupId= */ GROUP_ID_2);
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+            new LocalDatastoreServiceTestConfig()
+                .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+            new LocalUserServiceTestConfig())
+          .setEnvEmail(USER_EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("gmail.com")
+          .setEnvAttributes(
+              new HashMap(
+                  ImmutableMap.of(
+                      "com.google.appengine.api.users.UserService.user_id_key", USER_ID)));
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private DatastoreService datastore;
+  private CreateGroupServlet createGroupServlet;
+
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+    addUserToDatastore(datastore, USER_1);
+
+    // Set up a fake HTTP response.
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+    createGroupServlet = new CreateGroupServlet();
+  }
+
+  @After
+  public void tearDown() {
+    helper.setEnvIsLoggedIn(true);
+    helper.tearDown();
+  }
+
+  @Test
+  public void doPost_addGroup() throws Exception {
+    when(mockRequest.getParameter("groupName")).thenReturn(GROUP_NAME);
+
+    createGroupServlet.doPost(mockRequest, mockResponse);
+    Key groupKey = KeyFactory.createKey("Group", GROUP_ID);
+    Entity group = datastore.get(groupKey);
+    Key userKey = KeyFactory.createKey("User", USER_ID);
+    Entity user = datastore.get(userKey);
+
+    String jsonDs = new Gson().toJson(Group.fromEntity(group));
+    String jsonCurrent = new Gson().toJson(GROUP_1);
+    
```

Comment:
I'd move this new line and call the `String jsonDs` and below the `//Assert` block